File: tessoku/B04 - Binary Representation 2/B04.cs
using System;
using System.Linq;

class Program
{
   static void Main()
   {
       int[] N = Console.ReadLine().Select(c => c - '0').ToArray();

       long result = 0;
       for (int i = 0; i < N.Length; i++)
       {
           long power = (long)Math.Pow(2, N.Length - (i + 1));
           result += power * N[i];
       }

       Console.WriteLine(result);
   }
}

File: tessoku/B04 - Binary Representation 2/B04.php
<?php
$input = trim(fgets(STDIN));
$N = array_map('intval', str_split($input));
$result = 0;

for ($i = 0; $i < count($N); $i++) {
    $power = pow(2, count($N) - ($i + 1));
    $result += ($power * $N[$i]);
}

echo $result . "\n";